cli/launchpad/root.go

```diff
@@ -0,0 +1,85 @@
+package launchpad
+
+import (
+	"errors"
+	"fmt"
+	"gopkg.in/yaml.v2"
+	"log"
+)
+
+//go:generate go run static/includestatic.go
+
+// crdKind is the CustomResourceDefinition (CRD) which is indicated YAML's Kind value.
+type crdKind string
+
+// outputFlavor defines launchpad's generated output language.
+type outputFlavor string
+
+// Supported crdKind and outputFlavor.
+const (
+	KindCloudFoundation crdKind      = "CloudFoundation"
+	KindFolder          crdKind      = "Folder"
+	KindOrganization    crdKind      = "Organization"
+	outDm               outputFlavor = "dm"
+	outTf               outputFlavor = "tf"
+)
+
+// gState is a global scoped state to facilitate evaluation and output generation.
+var gState globalState
+
+// init initialize tracking for evaluated objects
+func init() {
+	gState.references = make(map[string]*parentRefYAML)
+	gState.evaluated.folders.YAMLs = make(map[string]*folderSpecYAML)
+}
+
+// NewGenerate takes file patterns as input YAMLs and output Infrastructure as
+// Code ready scripts based on specified output flavor.
+//
+// NewGenerate can be triggered by
+//   $ cft launchpad generate
+func NewGenerate(rawFilepath []string, outFlavor string, outputDir string) {
```

Comment:
I don't personally see the need for defining `type outputFlavor` above, string is probably fine.

But if you define it then you should use it. You're taking `outFlavor` as string here, when it probably should be `outputFlavor`.